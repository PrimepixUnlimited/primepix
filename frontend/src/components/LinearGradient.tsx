import React, { FC, ReactNode } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

interface Props {
  colors?: string[]
  children: ReactNode
  start: { x: number; y: number }
  end: { x: number; y: number }
}

const LinearGradientWrapper = ({
  children,
  colors = ['#4c669f', '#3b5998', '#192f6a']
}) => {
  return <LinearGradient colors={colors}>{children}</LinearGradient>
}

export default LinearGradientWrapper

const styles = StyleSheet.create({
  container: {}
})
