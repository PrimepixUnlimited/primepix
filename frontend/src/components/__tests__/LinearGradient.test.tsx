// components/__tests__/Hello.tsx
import "react-native";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { Text } from "react-native";

import LinearGradient from "../LinearGradient";

describe("LinearGradient snapshot", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <LinearGradient>
          <Text>Test</Text>
        </LinearGradient>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
