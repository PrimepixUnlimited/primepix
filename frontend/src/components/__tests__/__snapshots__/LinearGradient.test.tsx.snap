// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearGradient snapshot renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4283197087,
      4282079640,
      4279840618,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
>
  <Text>
    Test
  </Text>
</BVLinearGradient>
`;
